src/axom/mir/examples/tutorial_simple/mir_tutorial_simple.cpp

```diff
@@ -0,0 +1,287 @@
+  bool m_verbose {false};
+  std::string m_output_dir {};
+  RuntimePolicy m_policy {RuntimePolicy::seq};
+  std::string m_annotationMode {"report"};
+  axom::CLI::App m_app {};
+
+  /// Parse command line.
+  int parse(int argc, char **argv)
+  {
+    m_app.add_option("--test-case", m_test_case)
+      ->description("Select the test case.");
+
+    m_app.add_option("--output-dir", m_output_dir)
+      ->description("The directory for output files");
+
+    m_app.add_option("--iter-count", m_iter_count)
+      ->description("The number of iterations for MIR");
+
+    m_app.add_option("--iter-percent", m_iter_percent)
+      ->description("The percent error for iterative MIR");
+
+    m_app.add_flag("--verbose", m_verbose)->description("Verbose output");
+
+#if defined(AXOM_USE_CALIPER)
+    m_app.add_option("--caliper", m_annotationMode)
+      ->description(
+        "caliper annotation mode. Valid options include 'none' and 'report'. ")
+      ->capture_default_str()
+      ->check(axom::utilities::ValidCaliperMode);
+#endif
+
+    std::stringstream pol_sstr;
+    pol_sstr << "Set MIR runtime policy.";
+#if defined(AXOM_USE_RAJA) && defined(AXOM_USE_UMPIRE)
+    pol_sstr << "\nSet to 'seq' or 0 to use the RAJA sequential policy.";
+  #ifdef AXOM_USE_OPENMP
+    pol_sstr << "\nSet to 'omp' or 1 to use the RAJA OpenMP policy.";
+  #endif
+  #ifdef AXOM_USE_CUDA
+    pol_sstr << "\nSet to 'cuda' or 2 to use the RAJA CUDA policy.";
+  #endif
+  #ifdef AXOM_USE_HIP
+    pol_sstr << "\nSet to 'hip' or 3 to use the RAJA HIP policy.";
+  #endif
+#endif
+    m_app.add_option("-p, --policy", m_policy, pol_sstr.str())
+      ->capture_default_str()
+      ->transform(
+        axom::CLI::CheckedTransformer(axom::runtime_policy::s_nameToPolicy));
+
+    // Parse command line options.
+    try
+    {
+      m_app.parse(argc, argv);
+    }
+    catch(const axom::CLI::ParseError &e)
+    {
+      return m_app.exit(e);
+    }
+
+    int retval = 0;
+    checkTestCase(retval);
+    checkOutputDir();
+    checkIterationParams(retval);
+    return retval;
+  }
+
+  bool shouldIterate() const { return m_should_iterate; }
+  int numIterations() const { return m_iter_count; }
+  int iterPercentage() const { return m_iter_percent; }
+
+private:
+  void checkTestCase(int &retval)
+  {
+    if(m_test_case < 1 || m_test_case > 6)
+    {
+      retval = -1;
+      SLIC_ERROR("Invalid test case " << m_test_case);
+    }
+  }
+
+  void checkOutputDir()
+  {
+    if(!fs::pathExists(m_output_dir))
+    {
+      fs::makeDirsForPath(m_output_dir);
+    }
+  }
+
+  void checkIterationParams(int &retval)
+  {
+    if(m_should_iterate)
+    {
+      if(m_iter_count < 1)
+      {
+        retval = -2;
+        SLIC_ERROR("Invalid iteration count " << m_iter_count);
+      }
+
+      if(m_iter_percent <= 0. || m_iter_percent > 1.)
+      {
+        retval = -3;
+        SLIC_ERROR("Invalid iteration percentage " << m_iter_percent);
+      }
+    }
+  }
```

Comment:
Note -- I think these input validation checks can all be done w/ CLI11

Reply:
I removed this code and used CLI11 checking.